packages/nextjs/app/page.tsx

```diff
@@ -2,73 +2,222 @@
+  <div className={`rounded-lg border bg-card text-card-foreground shadow-sm mt-6 max-w-md mx-auto ${className}`}>
+    {children}
+  </div>
+);
+
+const CardHeader = ({ children, className = "" }: Props) => (
+  <div className={`flex flex-col space-y-1.5 p-6 ${className}`}>{children}</div>
+);
+
+const CardTitle = ({ children }: Props) => (
+  <h3 className="text-xl font-semibold leading-none tracking-tight">{children}</h3>
+);
+
+const CardContent = ({ children, className = "" }: Props) => <div className={`p-6 pt-0 ${className}`}>{children}</div>;
+
+const StatusSection = ({ connectedAddress, isAllowListed, isCheckedIn, isLoading }: BatchUserStatus) => {
+  if (isLoading) {
+    return (
+      <Card>
+        <CardContent>
+          <div className="flex items-center justify-center py-6">
+            <div className="flex items-center space-x-2">
+              <div className="h-6 w-6 animate-spin rounded-full border-b-2 border-gray-900"></div>
+              <p className="text-sm text-muted">Checking your status...</p>
+            </div>
+          </div>
+        </CardContent>
+      </Card>
+    );
+  }
+
+  if (!connectedAddress) {
+    return (
+      <Alert className="border-blue-200">
+        <AlertTitle>Connect Your Wallet</AlertTitle>
+        <AlertDescription>Please connect your wallet to view your Batch 10 status</AlertDescription>
+      </Alert>
+    );
+  }
+
+  if (!isAllowListed) {
+    return (
+      <Card>
+        <CardHeader className="pb-3">
+          <div className="flex items-center justify-between">
+            <CardTitle>Your Status</CardTitle>
+            <span className="rounded-full px-2 py-1 text-sm font-medium bg-red-100 text-red-800">Not a Member</span>
+          </div>
+        </CardHeader>
+        <CardContent>
+          <p className="text-sm opacity-70">You are not currently a member of Batch 10</p>
+        </CardContent>
+      </Card>
+    );
+  }
+
+  if (!isCheckedIn) {
+    return (
+      <Card>
+        <CardHeader className="pb-3">
+          <div className="flex items-center justify-between">
+            <CardTitle>Your Status</CardTitle>
+            <span className="rounded-full px-2 py-1 text-sm font-medium bg-yellow-100 text-yellow-800">
+              Ready to Check In
+            </span>
+          </div>
+        </CardHeader>
+        <CardContent>
+          <p className="text-sm opacity-70">You are a Batch 10 member! Deploy your contract to check in</p>
+        </CardContent>
+      </Card>
+    );
+  }
+
+  return (
+    <Card>
+      <CardHeader className="pb-3">
+        <div className="flex items-center justify-between">
+          <CardTitle>Your Status</CardTitle>
+          <span className="rounded-full px-2 py-1 text-sm font-medium bg-green-100 text-green-800">Checked In</span>
+        </div>
+      </CardHeader>
+      <CardContent>
+        <p className="text-sm opacity-70">
+          Active with wallet address: <Address address={connectedAddress} />
+        </p>
```

Comment:
There’s a hydration error occurring. Please make sure to always check your console for any potential errors. 
<img width="482" alt="Screenshot 2024-11-03 at 10 41 45" src="https://github.com/user-attachments/assets/ed5de89c-8c01-406b-9dcf-2a18d868290b">